common/components/forms/ImageCropUploadFormElement.jsx

```diff
@@ -37,27 +40,41 @@ class ImageCropUploadFormElement extends React.PureComponent<Props, State> {
     }
   }
 
+  static getStores(): $ReadOnlyArray<FluxReduceStore> {
+    return [FormFieldsStore];
+  }
+
+  static calculateState(prevState: State, props: Props): State {
+    let state: State = _.clone(prevState) || {};
+    state.currentImage = FormFieldsStore.getFormFieldValue(props.form_id);
+    return state;
+  }
+
   updateFormFields(fileInfo: FileInfo): void {
     this.refs.hiddenFormField.value = JSON.stringify(fileInfo);
     this.setState(
       { currentImage: fileInfo },
       () => this.props.onUpdate && this.props.onUpdate(fileInfo)
     );
+    UniversalDispatcher.dispatch({
+      type: "UPDATE_FORM_FIELD",
+      fieldName: this.props.form_id,
+      fieldValue: fileInfo,
+    });
+    // TODO: Need this?
     this.props.onSelection && this.props.onSelection(fileInfo);
   }
 
   render(): React$Node {
-    const previewImage = this.props.currentImage
-      ? this.props.currentImage.publicUrl
-      : "";
     return (
       <div>
         <ImageCropUploadButton
-          currentImage={previewImage}
+          currentImage={this.state.currentImage}
           aspect={this.props.aspect}
           buttonText={this.props.buttonText || "Upload Image"}
           onFileUpload={this._handleFileSelection.bind(this)}
         />
+        {/*TODO: Use HiddenFormField component*/}
```

Comment:
refactor to use the HiddenFormField component

Reply:
Added to list